### ISSUE TITLE
(German)(de-de) play next/previous song not working 

### ISSUE THREAD

**Issue by Loopinglu** [forslund/spotify-skill#140]
When I say "Spiele das nächste Lied" it only tells me which song is currently playing instead of switching to the next one.
When I say "Spiele das vorherige Lied" it start to play some news radio station. He does that actually quite often, even when I just want him to play something from phil collins "spiele etwas von phil collins". It's different radio stations though haha

**Comment by forslund** [comment]
Hi, it's actually not the spotify skill that is responsible for the next/previous commands. it uses the [mycroft-playback-control](https://github.com/mycroftai/skill-playback-control) to handle the common previous / next track commands. Check the translation of that skill (looks like next is translated as "nächster" only maybe it should be changed to `(nächste|nächtster|nächtstes)` or something.

Is there a special radio station that is picked up by "spiele etwas von phil collins"? or is it the default station?

**Comment by Loopinglu** [comment]
alright, so how come it does not recognise the command as aiming for raspotify but for this radio-thing?

Can I just create alternative commands by writing the words in brackets and seperate them by a `| ` ?

And no, it feels like it's always another radio station...

**Comment by forslund** [comment]
So this a bit hard to say for sure. The command generates a confidence based on the utterance, if it can detect that you asks for an album or artist or track it will report a higher confidence than if it's a generic query "play phil collins" So here it may be an issue with the translation of the regex for determining that it's an artist query. Alternatively the News skill is over matching and reporting that it is _very_ sure that Phil Collins is a radio station :)

Yes you can create alternative keywords by using the parenthesis + pipe expansion (word1|word2|word3).

**Comment by forslund** [comment]
[Checking](https://pythex.org/?regex=(dem%20K%C3%BCnstler%7Cder%20Gruppe%7Cder%20Band%7C(etwas%7Cirgendetwas%7CZeug%7CMusik%7CLieder)%20(von%7Cvom))%20(%3FP%3Cartist%3E.%2B)&test_string=spiele%20etwas%20von%20phil%20collins&ignorecase=0&multiline=0&dotall=0&verbose=0) the german regex it looks ok. Tomorrow I'll switch my Mycroft over to german to see if I can get some more insight into the issue.

**Comment by forslund** [comment]
I've tried some today and it seems like the news skill is over-matching. I get my default news if I ask "play discover weekly" but if I say "play discover weekly on spotify" Spotify gets precedence.

**Comment by Loopinglu** [comment]
Yeah exactly if one adds "on Spotify" it works. So is there any way to stop this overmatching?

**Comment by forslund** [comment]
I'll be checking the news skill and see if I should boost the confidence here slightly or if we should decrease the confidence for the npr-news

**Comment by Loopinglu** [comment]
Any progress on this issue? How does one boost or decrease the confidence?

**Comment by forslund** [comment]
Sorry for the slow response, I think I can do an update to at least make this better. If a query has a low matching value it's reported as a GENERIC match. See [L444](https://github.com/forslund/spotify-skill/blob/20.02/__init__.py#L444)

would be good if you can try to change that line to read

```python
                        level = CPSMatchLevel.TITLE
```

and see if the issue goes away.

**Comment by Loopinglu** [comment]
I tried it but the radio skills still let's activated :(

**Comment by penrods** [comment]
Seems like this issue is in the generic "skill-playback-control".   It believe you are saying the equivalent of "play the next song" in German, correct?  The words "next" and "song" are matches in English for the generic control from .voc file.  However "nächste" and "Leid" are not in the German translations of these files:

Next.voc -- has "next" in English but German vocabulary is missing "nächste":
https://github.com/MycroftAI/skill-playback-control/blob/20.02/locale/en-us/Track.voc
https://github.com/MycroftAI/skill-playback-control/blob/20.02/locale/de-de/Track.voc

and Track.voc -- has "song" in English but German vocabulary is missing "Leid":
https://github.com/MycroftAI/skill-playback-control/blob/20.02/locale/en-us/Track.voc
https://github.com/MycroftAI/skill-playback-control/blob/20.02/locale/de-de/Track.voc

**Comment by Loopinglu** [comment]
Yeah well I already fixed that issue!

It's just this radio matching thing remaining and that my log-in data gets deleted every few hours...
